backend/baseclass.pm

```diff
@@ -238,6 +238,11 @@ sub run_capture_loop {
                     # Very high limits! On a working socket, the maximum hits per 10 seconds will be around 60.
                     # The maximum hits per 10 seconds saw on a half open socket was >100k
                     if (check_select_rate($buckets, $wait_time_limit, $hits_limit, fileno $fh)) {
+                        if ($self->ssh_half_open_expected) {
+                            bmwqemu::diag "expected half-open sockets in ssh";
+                            $self->stop_ssh_serial;
+                            last;
+                        }
```

Comment:
Hm, the indendation / complexity is IMHO too high now, too convoluted code. Also, I do not think that any "ssh" specific code should be hidden in here. How about just raising this event as a proper exception and handling it on a higher level in a specific way?

Reply:
Thanks @okurz ,
This code is in a while loop, Raising this event and handling on higher level may need more carefully .
can you give example ? thanks

Reply:
Look for https://github.com/os-autoinst/os-autoinst/search?q=Exception&unscoped_q=Exception , maybe that helps